kube/src/client/mod.rs

```diff
@@ -72,6 +73,12 @@ impl Client {
         let uri_str = self.cluster_url.join(pandq.as_str())?;
         //trace!("Sending request => method = {} uri = {}", parts.method, uri_str);
 
+        let mut headers = parts.headers;
+        // If we have auth headers set, make sure they are updated and attached to the request
+        if let Some(auth_header) = self.config.get_auth_header().await? {
+            headers.insert(reqwest::header::AUTHORIZATION, auth_header);
+        }
```

Comment:
So we basically try to refresh the token - if needed, by keeping an in-memory `Config` around. That seems sensible, if a bit awkward. Though it does not do what `kubectl` does of updating the `~/.kube/config` with the new token.

Reply:
As far as I understood looking at the documentation, I thought the exec plugin was supposed to retrieve the token and it doesn't necessarily update the kubeconfig. When I tested it with `kubectl` and a toy shell script plugin I wrote, this is what it looked like after:

```yaml
- name: minikube-exec
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args: null
      command: ./exec-plugin.sh
      env: null
```

If I am wrong, I can try writing it out to the kubeconfig instead

Reply:
Oh, right. I might have misunderstood then. Then don't worry, no need to change the behaviour.
If you have some links on the exec plugin that you can shove into the markdown as a reference that would be useful though!

Reply:
I am not quite sure where to put the link on exec plugins. Should it be in the top level markdown or in the crate documentation?

Reply:
Somewhere around the relevant new enum field in the crate documentation preferably.

Reply:
I guess the enum is private to the crate, but that's fine, it's still helpful to people reading the source.